ext/phar/phar_object.c

```diff
@@ -4383,8 +4384,9 @@ PHP_METHOD(Phar, extractTo)
 								return;
 						}
 						if (NULL == (entry = zend_hash_find_ptr(&phar_obj->archive->manifest, Z_STR_P(zval_file)))) {
-							zend_throw_exception_ex(phar_ce_PharException, 0,
-								"Phar Error: attempted to extract non-existent file \"%s\" from phar \"%s\"", Z_STRVAL_P(zval_file), phar_obj->archive->fname);
+							goto all_files; /* potential directory match */
+dropstop: 
```

Comment:
I'd much prefer to have the logic for extracting a directory factored out into a function than doing these goto acrobatics :)

Reply:
I agree: the logic _should_ be factored out. I thought to go with this gymnastic (-:) approach because (a) existing code already set the `goto` precedent, (b) the refactoring would cloud the clarity of a bug fix diff, and (c) after tackling the ~40 other phar bugs I think a wholistic refactoring would be more efficient overall.

Still, I'm happy to refactor if you feel it's important to do so. I'd started to initially, but gave it thought and talked myself out of it per above.

Reply:
The problem here is not so much the goto itself, as the goto jumping back into a loop, essentially reimplementing a function call using gotos... I'd really like to avoid that. Incidentally, PHP itself prohibits such gotos at compile time :)